bitcode/devicelib.cl

```diff
@@ -125,6 +133,22 @@ double OVLD native_##NAME(double x, double y); \
 EXPORT float CL_NAME2(NAME##_native, f)(float x, float y) { return native_##NAME(x, y); } \
 EXPORT double CL_NAME2(NAME##_native, d)(double x, double y) { return native_##NAME(x, y); }
 
+#else
+
+#define DEF_OPENCL1F_NATIVE(NAME) \
+float OVLD native_##NAME(float x); \
+double OVLD native_##NAME(double x); \
+EXPORT float CL_NAME2(NAME##_native, f)(float x) { return NAME(x); } \
+EXPORT double CL_NAME2(NAME##_native, d)(double x) { return NAME(x); }
+
+#define DEF_OPENCL2F_NATIVE(NAME) \
+float OVLD native_##NAME(float x, float y); \
+double OVLD native_##NAME(double x, double y); \
+EXPORT float CL_NAME2(NAME##_native, f)(float x, float y) { return NAME(x, y); } \
+EXPORT double CL_NAME2(NAME##_native, d)(double x, double y) { return NAME(x, y); }
+
+#endif
```

Comment:
`devicelib.cl` contains implementations whereas all the mappings take place under hip/include/devicelib

so it would make more sense to me to have both the portable and non-portable implementations and then use the `CHIP_NONPORTABLE_MATH_INTRISINCS` define to map to the desired device function.